### ISSUE TITLE
sparse python interface

### ISSUE THREAD

**Issue by ichumuh** [qpSWIFT/qpSWIFT#10]
Let me start off by saying that your solver is great! I've tried a couple qp solvers for my motion planner project https://github.com/SemRoCo/giskardpy and qpSWIFT seems to be the fastest for my use case.

However, I currently have to waste a lot of computation time by having to work with dense matrices for qpSWIFT. It would be even faster if the python interface could accept scipy sparse matrices. In your readme you are saying that qpSWIFT is a sparse solver so I assume it should be possible?
Is such a feature planned?

**Comment by abhishek-pandala** [comment]
Thank you for your interest in our solver. Yes, we do have a rollout update planned for the sparse qpSWIFT python interface. This might take a while. But I see no reason why converting dense matrices to sparse matrices takes up a lot of time unless the matrices themselves are huge. I suggest profiling your code to see if this is actually the case.

**Comment by ichumuh** [comment]
Yeah, I've since realized that working with sparse matrices and only converting to dense at the end for qpSWIFT is still faster than working with dense matrices from the beginning. So this update would be less of a gain for me now. 
However, my matrices are large enough for the sparse->dense conversion to cost ~0.2ms per call and a similar amount would presumable be required on the solver side to turn it back into sparse. This is ~10% of the time it takes to solve the qp for me. I'm therefore still looking forward to the update.
